src/databrowser.js

```diff
@@ -32,7 +32,7 @@ define([
             head = pane.find(".fd-head-external-sources");
             headHeight = head.outerHeight(true) || 0;
             pane.data("min-size", headHeight)
-                .height(tree.height() * DATABROWSER_HEIGHT)
+                .height(Math.min(tree.height() * DATABROWSER_HEIGHT, headHeight * 12))
```

Comment:
This makes the default data browser height whichever is smaller: 33% of the tree height or 300px (25px * 12). On very small screens (height < 900px) this will choose 33% and on larger screens (height > 900px) it will choose 300px. Did I get that right?